test/integration/sapi-util.h

```diff
@@ -35,6 +35,16 @@
 #define TSS2_APP_ERROR(base_rc)   (TSS2_APP_ERROR_LEVEL | base_rc)
 #define TSS2_APP_RC_BAD_REFERENCE  TSS2_APP_ERROR (TSS2_BASE_RC_BAD_REFERENCE)
 /*
+ */
+#define TSS2_RETRY_EXP(expression)                         \
+    ({                                                     \
+        TSS2_RC __result = 0;                              \
+        do {                                               \
+            __result = (expression);                       \
+        } while ((__result & 0x0000ffff) == TPM_RC_RETRY); \
```

Comment:
There's no predefined mask for getting out the retry code?

Reply:
I had expected to get beat up for the empty comment above the macro (didn't see this till I made the PR). And the headers we ship now don't have a mask for this. There's likely one in the next iteration of the specs working their way through TCG. If not we'll get one added. I've got a TODO for making an RC decoding library like `strerror` (that should have a function to extract the base RC) I just haven't had the cycles to whip up a prototype yet.